File: leetcode/0062.Unique-Paths/UniquePaths.py
from functools import lru_cache

class Solution:
    # Runtime: 16 ms, faster than 99.90% of Python3 online submissions for Unique Paths.
    @lru_cache(maxsize=None)
    def uniquePaths(self, m: int, n: int) -> int:
        if m == 1 or n == 1:
            return 1

        return self.uniquePaths(m - 1, n) + self.uniquePaths(m, n - 1)

File: leetcode/0062.Unique-Paths/UniquePaths.go
package leetcode

// GolabalCache cache for uniquePaths
var GolabalCache = make(map[[2]int]int)

// Runtime: 0 ms, faster than 100.00% of Go online submissions for Unique Paths.
func uniquePaths(m int, n int) int {
	tup := [2]int{m, n}
	if res, isOK := GolabalCache[tup]; isOK {
		return res
	} else if m == 1 || n == 1 {
		return 1
	}
	result := uniquePaths(m-1, n) + uniquePaths(m, n-1)
	GolabalCache[tup] = result
	return result
}
